example/gluon/urban_sounds/datasets.py

```diff
@@ -0,0 +1,173 @@
+            raise RuntimeError("format {} not supported currently.".format(file_format))
+        if skip_header:
+            skip_rows = 1
+        else:
+            skip_rows = 0
+        self._list_audio_files(self._root, skip_rows=skip_rows)
+
+
+    def _list_audio_files(self, root, skip_rows=0):
+        """Populates synsets - a map of index to label for the data items.
+        Populates the data in the dataset, making tuples of (data, label)
+        """
+        self.synsets = []
+        self.items = []
+        if self._train_csv is None:
+            for folder in sorted(os.listdir(root)):
+                path = os.path.join(root, folder)
+                if not os.path.isdir(path):
+                    warnings.warn('Ignoring {}, which is not a directory.'.format(path))
+                    continue
+                label = len(self.synsets)
+                self.synsets.append(folder)
+                for filename in sorted(os.listdir(path)):
+                    file_name = os.path.join(path, filename)
+                    ext = os.path.splitext(file_name)[1]
+                    if ext.lower() not in self._exts:
+                        warnings.warn('Ignoring {} of type {}. Only support {}'\
+                        .format(filename, ext, ', '.join(self._exts)))
+                        continue
+                    self.items.append((file_name, label))
+        else:
+            skipped_rows = 0
+            with open(self._train_csv, "r") as traincsv:
+                for line in traincsv:
+                    skipped_rows = skipped_rows + 1
+                    if skipped_rows <= skip_rows:
+                        continue
+                    filename = os.path.join(root, line.split(",")[0])
+                    label = line.split(",")[1].strip()
+                    if label not in self.synsets:
+                        self.synsets.append(label)
+                    if self._format not in filename:
+                        filename = filename+self._format
+                    self.items.append((filename, nd.array([self.synsets.index(label)]).reshape((1,))))
+
+            #Generating the synset.txt file now
+            if not os.path.exists("./synset.txt"):
+                with open("./synset.txt", "w") as synsets_file:
+                    for item in self.synsets:
+                        synsets_file.write(item+os.linesep)
+                print("Synsets is generated as synset.txt")
+            else:
+                warnings.warn("Synset file already exists in the current directory! Not generating synset.txt.")
+
+
+    def __getitem__(self, idx):
+        """Retrieve the item (data, label) stored at idx in items"""
+        filename = self.items[idx][0]
+        label = self.items[idx][1]
+        if librosa is not None:
+            X1, _ = librosa.load(filename, res_type='kaiser_fast')
+            return nd.array(X1), label
+        else:
+            warnings.warn(" Dependency librosa is not installed! \
```

Comment:
Can it continue safely if librosa is not installed? Will this case be valid if we throw an exception during import